From: Fangzhou Chen <fangzhou@fydeos.io>
Date: 2025-03-17T06:09:18
Subject: keep factory install dlc image and remove encrypted.block when clobber
Signed-off-by: Fangzhou Chen <fangzhou@fydeos.io>

keep factory install dlc image when powerwash(safe wipe) or dev clobber

powerwash command from chrome and session manager equals

```
echo 'fast safe keepimg preserve_lvs reason=some_reason' > /mnt/stateful_partition/factory_install_reset
```

dev clobber commajnd is `echo 'clobber' > /mnt/stateful_partition/.update_available`,
this patch keeps the behavior of removing `encrypted.block` and
`encrypted.key`, to reduce confusion

keep tpm2-simulator data for dev clobber, removing these data will cause tpm errors


Index: init/clobber/clobber_state.cc
===================================================================
--- init.orig/clobber/clobber_state.cc
+++ init/clobber/clobber_state.cc
@@ -517,7 +517,7 @@ std::vector<base::FilePath> ClobberState
           args_.safe_wipe, args_.ad_migration_wipe, args_.rollback_wipe,
           args_.rma_wipe, debug_build == 1, false);
 
-  if (args_.factory_wipe || args_.default_key_migration_wipe) {
+  if (args_.factory_wipe || args_.default_key_migration_wipe || args_.safe_wipe) {
     for (auto& path :
          libpreservation::GetFactoryPreservationPathList(stateful_)) {
       stateful_paths.insert(path);
Index: init/startup/stateful_mount.cc
===================================================================
--- init.orig/startup/stateful_mount.cc
+++ init/startup/stateful_mount.cc
@@ -606,8 +606,8 @@ bool StatefulMount::DevUpdateStatefulPar
     std::vector<base::FilePath> preserved_paths = {
         stateful_.Append(kLabMachine),
         stateful_.Append(kDevModeFile),
-        stateful_.Append("encrypted.block"),
-        stateful_.Append("encrypted.key"),
+        stateful_.Append("unencrypted/dlc-factory-images"),
+        stateful_.Append("unencrypted/tpm2-simulator"),
         stateful_.Append(kUnencrypted).Append(kClobberLogFile),
         stateful_.Append(kUnencrypted).Append(kClobberStateLogFile),
         stateful_.Append(kUnencrypted).Append(kDevImageBlockFile),
